bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java

```diff
@@ -109,22 +111,36 @@ public GroupFunction getFunction() {
      * @return all members of this and all contained {@link GroupItem}s
      */
     public Set<Item> getAllMembers() {
-        Set<Item> allMembers = new HashSet<Item>();
-        collectMembers(allMembers, members);
+        Set<Item> allMembers = getMembers((Item i) -> !(i instanceof GroupItem));
         return ImmutableSet.copyOf(allMembers);
     }
 
     private void collectMembers(Set<Item> allMembers, Set<Item> members) {
         for (Item member : members) {
             if (member instanceof GroupItem) {
                 collectMembers(allMembers, ((GroupItem) member).members);
+                allMembers.add(member);
+

```

Comment:
No, this unfortunately just happens to me sometimes. I create them to get a better overview while coding and moving lines around and then forget to clean them up afterwards. I have removed it now.